src/list/src/rich_listScript.sml

```diff
@@ -1178,6 +1178,17 @@ val TAKE_SNOC = Q.store_thm ("TAKE_SNOC",
    THEN RES_TAC
    THEN ASM_REWRITE_TAC []);
 
+Theorem TAKE_FRONT :
+    !l n. l <> [] /\ n <= PRE (LENGTH l) ==> TAKE n (FRONT l) = TAKE n l
```

Comment:
Rather than `n <= PRE ...`, please use `n < ...`.

Reply:
Yes, somehow I overlooked the obvious fact that `n <= PRE (LENGTH l) <=> n < LENGTH l`...